test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperDescriptorFactoryTest.cs

```diff
@@ -324,41 +541,347 @@ public void CreateDescriptor_OverridesTagNameFromAttribute()
-                    AssemblyName),
-                new TagHelperDescriptor(
-                    "p",
-                    typeof(MultipleAttributeTagHelper).FullName,
-                    AssemblyName),
-                new TagHelperDescriptor(
-                    "div",
-                    typeof(MultipleAttributeTagHelper).FullName,
-                    AssemblyName)
-            };
+            var errorSink = new ParserErrorSink();
+            var attribute = new TargetElementAttribute(name);
 
             // Act
-            var descriptors = TagHelperDescriptorFactory.CreateDescriptors(AssemblyName,
-                typeof(MultipleAttributeTagHelper));
+            TagHelperDescriptorFactory.ValidTargetElementAttributeNames(attribute, errorSink);
 
             // Assert
-            Assert.Equal(expectedDescriptors, descriptors, CompleteTagHelperDescriptorComparer.Default);
+            var errors = errorSink.Errors.ToArray();
+            for (var i = 0; i < errors.Length; i++)
+            {
+                Assert.Equal(expectedErrorMessages[i], errors[i].Message);
+                Assert.Equal(SourceLocation.Zero, errors[i].Location);
+            }
+        }
+
+        public static TheoryData ValidNameData
+        {
+            get
+            {
+                // name, expectedNames
+                return new TheoryData<string, IEnumerable<string>>
+                {
+                    { "p", new[] { "p" } },
+                    { " p", new[] { "p" } },
+                    { "p ", new[] { "p" } },
+                    { " p ", new[] { "p" } },
+                    { "p,div", new[] { "p", "div" } },
+                    { " p,div", new[] { "p", "div" } },
+                    { "p ,div", new[] { "p", "div" } },
+                    { " p ,div", new[] { "p", "div" } },
+                    { "p, div", new[] { "p", "div" } },
+                    { "p,div ", new[] { "p", "div" } },
+                    { "p, div ", new[] { "p", "div" } },
+                    { " p, div ", new[] { "p", "div" } },
+                    { " p , div ", new[] { "p", "div" } },
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidNameData))]
+        public void GetCommaSeparatedValues_OutputsCommaSeparatedListOfNames(
+            string name,
+            IEnumerable<string> expectedNames)
+        {
+            // Act
+            var result = TagHelperDescriptorFactory.GetCommaSeparatedValues(name);
+
+            // Assert
+            Assert.Equal(expectedNames, result);
+        }
+
+        [Fact]
+        public void GetCommaSeparatedValues_OutputsEmptyArrayForNullValue()
+        {
+            // Act
+            var result = TagHelperDescriptorFactory.GetCommaSeparatedValues(text: null);
+
+            // Assert
+            Assert.Empty(result);
+        }
+
+        [TargetElement(Attributes = "class")]
+        private class AttributeTargetingTagHelper : TagHelper
```

Comment:
`OverrideNameTagHelper` is only there to ensure we override the naming convention of `TagHelper`s when `TargetElement` is present. This is only targeting attributes which have no convention.